api/v1alpha1/httproute_types.go

```diff
@@ -264,6 +264,14 @@ type ForwardToTarget struct {
 	//
 	// +optional
 	TargetPort *TargetPort `json:"targetPort" protobuf:"bytes,2,opt,name=targetPort"`
+
+	// Filter defines filters to apply to the request before forwarding it to
+	// a targetRef.
+	//
+	// Support: Core
+	//
+	// +optional
+	Filter *HTTPRouteFilter `json:"filter" protobuf:"bytes,3,opt,name=filter"`
```

Comment:
We should consider renaming `HTTPRouteFilter` to `HTTPRequestFilter` or make it as simple as `HTTPFilter`, which can be used as an extension point to add in routing functionality like caching, authentication, complex transformations, etc.

Reply:
There can also be filter to add/remove headers from the response prior to sending the response to the client. Shall that be extension?

Reply:
@yiyangy Great timing. I was actually thinking about this right now and was considering opening an issue to discuss exactly this. Keeping it in extension seems reasonable to me.

Reply:
@yiyangy thanks for the review and for bringing response filtering to my attention. I would like to separate request and response handling, so please create an issue and link to this PR.

Reply:
@hbagdi commit `53eaea7` replaces `Route` with `Request` in `HTTPRoute` match, filter and action types. I decided on `Request` based on possibly supporting `Response` types in the future.